compiler/src/dotty/tools/dotc/core/TypeComparer.scala

```diff
@@ -666,17 +666,10 @@ class TypeComparer(@constructorOnly initctx: Context) extends ConstraintHandling
                 isSubType(info1, info2)
 
             if defn.isFunctionType(tp2) then
-              if defn.isPolyFunctionType(tp2) then
-                // TODO should we handle ErasedFunction is this same way?
-                tp1.member(nme.apply).info match
-                  case info1: PolyType =>
-                    return isSubInfo(info1, tp2.refinedInfo)
-                  case _ =>
-              else
-                tp1w.widenDealias match
-                  case tp1: RefinedType =>
-                    return isSubInfo(tp1.refinedInfo, tp2.refinedInfo)
-                  case _ =>
+              tp1w.widenDealias match
+                case tp1: RefinedType =>
```

Comment:
What if the refinement is for another member than `apply` ? I don't really understand either the previous or current logic,  it seems we should be checking for `apply` in both `tp1` and `tp2`, but then why do we even need special logic for functions instead of the normal logic for refinements?

Reply:
This code was added in https://github.com/lampepfl/dotty/pull/15877/files#diff-c163f9660fb06bb487b0120921d06353fc52154ae1b3066f58da1c73ecd00914R613-R641

It has the following comment
```
// A relaxed version of subtyping for dependent functions where method types
// are treated as contravariant.
// TODO: Merge with isSubInfo in hasMatchingMember. Currently, we can't since
// the isSubinfo of hasMatchingMember has problems dealing with PolyTypes
// (---> orphan params during pickling)
```

Removing this completely makes some cc tests fail (such as tests/neg-custom-args/captures/heal-tparam-cs.scala).

@odersky do you have more insight on this?

Reply:
a orphan param thing can be fixed by using `comparingTypeLambdas` when recursing in the PolyTypes, I'll update the PR.